### PR TITLE
use obj.pk + patch all the time

### PR DESCRIPTION
**Pull request by m-vdb**

### CODE REVIEW

File: mongodbforms/documentoptions.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def __init__(self, document, meta=None):
         # if it doesn't have one it's an embedded document
         #if 'id_field' in self._meta:
         #    self.pk_name = self._meta['id_field']
-        self._init_pk()
+        # self._init_pk()
```
Comment (olivierlefloch) [inline mongodbforms/documentoptions.py:155]: Gné?

Comment (m-vdb) [inline mongodbforms/documentoptions.py:155]: i'm playing

File: mongodbforms/documentoptions.py [hunk 2]
```diff
@@ -197,15 +197,14 @@ def _init_pk(self):
...
 
-        def _get_pk_val(self):
-            # needs to be computed on the fly
-            pk_name = self._meta['id_field']  # default to id
-            return getattr(self, pk_name)
-        
+        def _get_pk_val(obj):
+            return obj.pk
+
+        patch_document(_get_pk_val, self.document)
```
Comment (olivierlefloch) [inline mongodbforms/documentoptions.py:203]: Why not a lambda?

File: mongodbforms/documentoptions.py (hunk 1 above)
Comment (olivierlefloch) [inline mongodbforms/documentoptions.py:155]: On `master`?

**Pull request merged**